.gitlab-ci.yml

```
@@ -73,15 +82,37 @@ variables:
 .docker-env:                       &docker-env
   image:                           "${CI_IMAGE}"
   before_script:
+    - *check-passed-job
     - *rust-info-script
   tags:
     - linux-docker
 
 .kubernetes-env:                   &kubernetes-env
   image:                           "${CI_IMAGE}"
+  before_script:
+    - *check-passed-job
   tags:
     - kubernetes-parity-build
 
+#### stage:                        prepare
+
+prepare:
+  stage:                           prepare
+  <<:                              *kubernetes-env
```

```suggestion
  <<:                              *docker-env
```
Should be executed on more powerful ci runners

---

Same response as https://github.com/paritytech/substrate/pull/11765#discussion_r931949160